File: backend/Backend/Models/Team.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel;

namespace ProfkomBackend.Models
{
    public enum MemberType
    {
        Aparat,
        Profburo,
        Viddil
    }

    public class Team
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Position { get; set; } = string.Empty;

        [Required]
        public MemberType Type { get; set; }

        public string? ImageUrl { get; set; }
        public string? Email { get; set; }

        [Required]
        public int OrderInd { get; set; }           // number у фронті

        [Required]
        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow; // created_at

        public bool IsChoosed { get; set; } = false;
    }
}

File: frontend/src/types/team.ts
export type MemberType = "Aparat" | "Profburo" | "Viddil";

export const MemberTypeDisplay: Record<MemberType, string> = {
  Aparat: "Член Апарату Профкому",
  Profburo: "Голова Профбюро",
  Viddil: "Голова Відділу",
};

export interface TeamFormData {
  name: string;
  position: string;
  type: number;
  orderInd: number;
  isActive: boolean;
  email?: string;
  imageUrl?: string;
}

export interface TeamMember {
  id: number;
  name: string;
  position: string;
  type: number;
  orderInd: number;
  isActive: boolean;
  createdAt: string;
  email?: string;
  imageUrl?: string;
  isChoosed: boolean;
}

export const APARAT_TYPE  = 0;
export const PROFBURO_HEAD_TYPE = 1;
export const VIDDIL_HEAD_TYPE  = 2;